Feature/overlay refactor

emclaug2 opened the pull request:
<!-- Include a bulleted list summarizing the main changes you have made in this PR -->
#### Changes proposed in this Pull Request:
- Use Heroes in the complex bottom sheet
- Add empty state placeholder

joebochill commented on src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.scss:
```diff
@@ -1,22 +1,19 @@
+}
+
+.active * {
+    color: #007bc1;
```
Can we pull this from the colors package

emclaug2 replied:
Absolutely :)

joebochill commented on src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.html:
```diff
@@ -1,65 +1,60 @@
+                    <mat-icon mat-list-icon [style.fontSize.px]="24">access_time</mat-icon>
                 </button>
-                <span mat-line style="display: block;">Time</span>
-            </a>
-            <a [class.active]="activeSort == filterTypes.EVENT_TYPE" (click)="sortData(filterTypes.EVENT_TYPE)">
-                <button mat-icon-button>
-                    <mat-icon data-cy="btm-sheet-type-btn" mat-list-icon>info</mat-icon>
+            </pxb-hero>
```
The whole hero should be clickable, rather than just the icon
- You'll probably want to convert those mat-icon-buttons to plain mat-icons when you make this change

emclaug2 replied:
I've made the entire Hero clickable; no more mat-buttons.

joebochill commented on src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.html:
```diff
@@ -1,65 +1,60 @@
-            <a [class.active]="activeSort == filterTypes.TIME" (click)="sortData(filterTypes?.TIME)">
-                <button mat-icon-button>
-                    <mat-icon mat-list-icon>access_time</mat-icon>
+        <div class="mat-h3">Sort By:</div>
```
I think you got the labels backwards...Huayun wants us to _use_ the Overline label style in both React and Angular.

emclaug2 replied:
Ah, okay! This is reverted now.

emclaug2 commented:
Once this PR gets merged I'll update React to match.

jeffvg commented:
The title here is in h3 tag but is rendering in all uppercase and only noticed because the cypress test is failing on it.

![image](https://user-images.githubusercontent.com/37150565/92495320-c4735c00-f1c4-11ea-98f9-786ca3e701a7.png)

emclaug2 commented:
@jeffvg  I've removed the  `text-transform: uppercase` from the overline typography.

This is a comparison between the two overlay examples right now:

![image](https://user-images.githubusercontent.com/6538289/92497511-6eec7e80-f1c7-11ea-9eb0-0326959b1d74.png)

I'm planning on updating React to do the following:
1. match angular tyopgraphy
2.  Use hero components
3.  Match container padding
4.  Update bottom button to have elevation

joebochill commented:
Both should be using the `overline` style with uppercase transform for the labels.

@huayunh I warned you that change was going to cause some confusion lol 😆

emclaug2 commented:
The heroes now look a bit spaced out, so the fix would be to apply a max-width to each hero.  I'm going to hold off on applying that style for now since the root class of the hero is going to be changed as a part of another PR.

jeffvg commented:
> @jeffvg I've removed the `text-transform: uppercase` from the overline typography.
> 
> This is a comparison between the two overlay examples right now:
> 
> React is on the left, angular is on the right.
> ![image](https://user-images.githubusercontent.com/6538289/92497511-6eec7e80-f1c7-11ea-9eb0-0326959b1d74.png)
> 
> I'm planning on updating React to do the following:
> 
> 1. match angular typography
> 2. Use hero components
> 3. Match overlay container padding
> 4. Update bottom button to have elevation

did that removal of text-transform: uppercase get pushed in?

emclaug2 commented:
> > @jeffvg I've removed the `text-transform: uppercase` from the overline typography.
> > This is a comparison between the two overlay examples right now:
> > React is on the left, angular is on the right.
> > ![image](https://user-images.githubusercontent.com/6538289/92497511-6eec7e80-f1c7-11ea-9eb0-0326959b1d74.png)
> > I'm planning on updating React to do the following:
> > 
> > 1. match angular typography
> > 2. Use hero components
> > 3. Match overlay container padding
> > 4. Update bottom button to have elevation
> 
> did that removal of text-transform: uppercase get pushed in?



> > @jeffvg I've removed the `text-transform: uppercase` from the overline typography.
> > This is a comparison between the two overlay examples right now:
> > React is on the left, angular is on the right.
> > ![image](https://user-images.githubusercontent.com/6538289/92497511-6eec7e80-f1c7-11ea-9eb0-0326959b1d74.png)
> > I'm planning on updating React to do the following:
> > 
> > 1. match angular typography
> > 2. Use hero components
> > 3. Match overlay container padding
> > 4. Update bottom button to have elevation
> 
> did that removal of text-transform: uppercase get pushed in?

Yep, the "Sort By" and "Show" text is now upper-cased.

joebochill commented:
> The heroes now look a bit spaced out, so the fix would be to apply a max-width to each hero. I'm going to hold off on applying that style for now since the root class of the hero is going to be changed as a part of another PR.

That max-width would go in the example though, not in the component, right? We don't know what kind of labels people will be using, so we should probably leave that up to them.

joebochill commented:
Refactor looks good. Just one question, should we be using black[500] for the hero icons (in this particular use case)?

emclaug2 commented:
> > The heroes now look a bit spaced out, so the fix would be to apply a max-width to each hero. I'm going to hold off on applying that style for now since the root class of the hero is going to be changed as a part of another PR.
> 
> That max-width would go in the example though, not in the component, right? We don't know what kind of labels people will be using, so we should probably leave that up to them.

You're right, I didn't make the change yet because the class names were going to change soon, but I can just apply css to the hero tags instead of the classes.

emclaug2 commented:
Here's the complex bottomsheet now:

![image](https://user-images.githubusercontent.com/6538289/92599934-e3332a80-f278-11ea-8024-9ac7ac83d933.png)

emclaug2 commented:
> Refactor looks good. Just one question, should we be using black[500] for the hero icons (in this particular use case)?

I'm not sure; right now it's using the default Hero colors which I think looks alright.

joebochill commented:
> > Refactor looks good. Just one question, should we be using black[500] for the hero icons (in this particular use case)?
> 
> I'm not sure; right now it's using the default Hero colors which I think looks alright.

I'm wondering because the other text in the bottomsheets is black[500]

@huayunh thoughts?

emclaug2 commented:
@joebochill  @huayunh 
Is the elevation above the cancel button area preferred over a flat divider?   This is only present on mobile viewports. 

![image](https://user-images.githubusercontent.com/6538289/92602928-bf71e380-f27c-11ea-9be9-1019e7ff5a12.png)

huayunh commented:
(1) Happy with the default hero

(2) the elevation is there so that the cancel button is pinned at the bottom, and we can scroll vertically when the content overflows. This overflow typically only occurs on a mobile device held at the landscape position.
